orderby-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/CompositeKey.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.aggregator;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.format.StructuredRecordStringConverter;
+import com.google.common.base.Joiner;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.io.WritableComparable;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * CustomWritable for the composite key.
+ */
+public class CompositeKey implements Writable, WritableComparable<CompositeKey> {
+
+  private static final String FIELD_DELIMITER = "->";
+  private final Gson gson = new GsonBuilder().create();
+  private final Type mapType = new TypeToken<LinkedHashMap>() {  }.getType();
+  private final Type schemaType = new TypeToken<Schema>() {  }.getType();
+  private String structureRecordJSON;
+  private String sortFieldsJSON;
+  private String schemaJSON;
+
+  public CompositeKey() {
+  }
+
+  public CompositeKey(Text structureRecordJSON, String sortFieldsJSON, String schemaJSON) {
+    this.structureRecordJSON = structureRecordJSON.toString();
+    this.sortFieldsJSON = sortFieldsJSON;
+    this.schemaJSON = schemaJSON;
+  }
+
+  public static int getSortingOrder(String order) {
+    if (order.equalsIgnoreCase("asc")) {
+      return 1;
+    } else {
+      return -1;
+    }
+  }
+
+  /**
+   * Get sorting order for structured records depending on the type of value contained in key
+   *
+   * @param key               contains the value to sort the data
+   * @param value             determines whether the objects are to be sorted in ascending or descending order for key
+   * @param structuredRecord1 Structured record
+   * @param structuredRecord2 Structured record to compare against
+   * @return sorting order of the complete structured record
+   */
+  public static int compareRecords(String key, String value, StructuredRecord structuredRecord1,
```

Comment:
Comparison has to support recursive complex records. So you might want to implement a recursive function for doing that. StructuredRecord can have basic types as well as Record, Union, etc.